packages/flutter/lib/src/foundation/assertions.dart

```diff
@@ -672,9 +672,7 @@ class FlutterErrorDetails with Diagnosticable {
     super.debugFillProperties(properties);
     final DiagnosticsNode verb = ErrorDescription('thrown${ context != null ? ErrorDescription(" $context") : ""}');
     final Diagnosticable? diagnosticable = _exceptionToDiagnosticable();
-    if (exception is NullThrownError) {
```

Comment:
This shouldn't be removed until the non-null-safe mode is removed.

Instead, can you add the ignore that was removed in https://github.com/flutter/flutter/pull/108915/files?

```dart
// ignore: deprecated_member_use
```

That way the tests below can remain unchanged as well.

Reply:
OK, we're going to totally remove these errors in the next 1-2 weeks (after main opens op after the branch cut), so I'll keep this PR for then, and open a new PR to add the `ignore`'s.

Reply:
https://github.com/flutter/flutter/pull/116135

Reply:
We're now past the Dart branch-point that removes non-null-safe-mode, so please PTAL again!

Reply:
Dart 3 has now rolled to the framework, so we should be ready to take this PR?

Reply:
Yeah, I think so. Yesterday, I left a review comment below. Could you address that? We should be good to go then.